### PR TITLE
Enabled eval_double_single_dispatch and added tests

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: src/eval_double.cpp [hunk 1]
```diff
@@ -377,14 +377,13 @@ std::vector<fn> init_eval_double()
 
 const static std::vector<fn> table_eval_double = init_eval_double();
 
-double eval_double(const Basic &b)
-{
-#if defined(CSYMPY_SINGLE_DISPATCH)
```
Comment (certik) [inline src/eval_double.cpp]: You should also remove the `CSYMPY_SINGLE_DISPATCH` definition above.

File: src/tests/eval/test_eval_double.cpp [hunk 2]
```diff
@@ -46,53 +46,72 @@ using CSymPy::acoth;
...
 
-    r3 = add(sinh(one), add(cosh(one), add(tanh(one), coth(one))));
-    assert(::fabs(eval_double(*r3) - 4.792911269914) < 1e-12);
+    vec.push_back(add(sinh(one), add(cosh(one), add(tanh(one), coth(one)))));
+    result.push_back(4.792911269914);
 
-    r3 = add(asinh(r5), add(acosh(r5), add(atanh(r4), acoth(r5))));
-    assert(::fabs(eval_double(*r3) - 5.010335118942) < 1e-12);
+    vec.push_back(add(asinh(r4), add(acosh(r4), add(atanh(r3), acoth(r4)))));
+    result.push_back(5.010335118942);
```
Comment (certik) [inline src/tests/eval/test_eval_double.cpp]: I wonder if you can use a vector of std::pair (expression, result), and then use initializers like:
```
vec = {
{tan(pow(r1, r2)), 1.314847038576},
{mul(r1, r2), 0.403422680111},
...
};
```

**Comment by certik** [comment]
Otherwise it looks good, thanks!

File: src/eval_double.cpp (hunk 1 above)
Comment (isuruf) [inline src/eval_double.cpp]: Done

**Comment by isuruf** [comment]
I pushed a commit with `std::pair`. I think it looks nice now.

**Comment by certik** [comment]
It looks good now. The test failure worries me, but it looks unrelated.  Let's see if the latest commit passes.

**Comment by isuruf** [comment]
Tests pass now. Test failure was because of an error in apt-get.

**Comment by isuruf** [comment]
Rebased on top of master. Same build failure on Shippable as #433

**Comment by certik** [comment]
Thanks for all the work @isuruf. +1 to merge.

**Pull request merged**